src/openrct2/world/Footpath.cpp

```diff
@@ -1598,27 +1598,29 @@ void footpath_queue_chain_push(uint8 rideIndex)
  */
 void footpath_update_queue_chains()
 {
-    for (uint8 *queueChainPtr = _footpathQueueChain; queueChainPtr < _footpathQueueChainNext; queueChainPtr++) {
-        uint8 rideIndex = *queueChainPtr;
-        Ride *ride = get_ride(rideIndex);
-        if (ride->type == RIDE_TYPE_NULL) {
+    for (uint8 * queueChainPtr = _footpathQueueChain; queueChainPtr < _footpathQueueChainNext; queueChainPtr++)
```

Comment:
This should use range-based `for` and there's little reason for `queueChainPtr` and `rideIndex` to coexist,

Reply:
Oh, it doesn't loop through all the elements, but only up to a pointer somewhere into this array. Nevermind then.